chapter/evaluation.md

```diff
@@ -37,10 +37,189 @@ In particular, it details the survey ([@sec:qualitative]) which was conducted wi
+Additionally record field completion was found to be missing, yet highly valued.
+
+#### Document Navigation
+
+Results and comments about the Go-To-Definition and Find-Referneces were polarized.
+On the one hand users reported no issues while others experienced unexpected behavior or were unabke to use the feature at all (cf. [@fig:results-comp-experience]).
+Similarly, the comments on one hand suggest that "the feature works well and is quick" while on the other mention inconsistencies and unavailabilty.
+More practically, cross file navigation was named an important missing feature.
+
+#### General Performance
+
+The responses to this item suggest that NLS' perfomance is largely dependent on its usage.
+On unmodified files queries were reported to evaluate "instantaniously".
+However modifying files caused that "modifications stack up" causing high CPU usage and generally "very slow" responses.
+Others pointed out that documentation was slow to resolve while the server itself was "generally fast".
+
+
 ### Quantitative
+
+The quantitative evaluation focuses on the performance characteristics of NLS.
+As described in [@sec:eval-methods-quantitative] a tracing module was embedded into the NLS binary which recorded the runtime together with the size of the analyzed data, i.e., the number of linearization items [@sec:linearization] or size of the analyzed file.
+
+#### Dataset
+
+The underlying data set consists of 16760 unique trace records.
+Since the `textDocument/didOpen` method is executed on every update of the source, it greatly outnumbers the other events.
+The final distrubution of methods traced is:
+
+| Method                    | count | linearization based |
+| ------------------------- | ----- | ------------------- |
+| `textDocument/didOpen`    | 13436 | no                  |
+| `textDocument/completion` | 2981  | yes                 |
+| `textDocument/hover`      | 227   | yes                 |
+| `textDocument/definition` | 68    | yes                 |
+| `textDocument/references` | 49    | yes                 |
+
+: Number of traces per LSP method
+
+![Distribution of linearization based LSP requests](log_analysis/figures/distribution-by-lin-size.svg){#fig:distribution-by-lin-size}
+
+![Distribution of file analysis requests](log_analysis/figures/distribution-by-file-size.svg){#fig:distribution-by-file-size}
+
+
+[Figures @fig:distribution-by-lin-size, @fig:distribution-by-file-size] break up these numbers by method and linearization size or file size respectively.
+The first figure shows a peak number of traces for completion events between $0$ to $1$ linearization items as well as local maxima around a linearization size of $20$ to $30$ and sustained usage of completion requests in files of $90-400$ items.
+Similar to the completion requests (but well outnumbered in total counts), other other methods were used mainly in the range between $200$ and $400$ linearization items.
```

Comment:
```suggestion
Similar to the completion requests (but well outnumbered in total counts), other methods were used mainly in the range between $200$ and $400$ linearization items.
```

Reply:
Maybe a reminder, even quick, about what exactly means "linearization size/linearization items" could be useful here. I don't know where this chapter will be in the final manuscript, but the reader may not necessarily have still a precise idea of what all of those are.